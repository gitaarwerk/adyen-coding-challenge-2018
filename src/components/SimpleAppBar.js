import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';

const appBarStyling = {
  position: 'fixed',
  top: 0,
  left: 0,
  width: '100%',
  background:
    'rgba(10,191,82,1) linear-gradient(45deg, rgba(10,191,82,1) 0%, rgba(106,208,155,1) 100%)'
};

const SimpleAppBar = () => (
  <AppBar position="static" style={appBarStyling}>
    <Toolbar variant="dense">
      <Typography variant="title" color="inherit">
        GoOut Adyen
      </Typography>
    </Toolbar>
  </AppBar>
);

export default SimpleAppBar;
