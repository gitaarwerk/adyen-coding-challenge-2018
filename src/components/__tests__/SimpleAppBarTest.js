import React from 'react';
import { shallow } from 'enzyme';

import SimpleAppBar from '../SimpleAppBar';

describe('Simple App Bar', () => {
  test('renders correctly', () => {
    const component = shallow(<SimpleAppBar />);

    expect(component).toMatchSnapshot();
  });
});
