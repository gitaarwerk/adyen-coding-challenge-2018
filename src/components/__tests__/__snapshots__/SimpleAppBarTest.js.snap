// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple App Bar renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SimpleAppBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(Toolbar)
        variant="dense"
>
        <WithStyles(IconButton)
                aria-label="Menu"
                color="inherit"
        >
                <pure(Menu) />
        </WithStyles(IconButton)>
        <WithStyles(Typography)
                color="inherit"
                variant="title"
        >
                GoOut Adyen
        </WithStyles(Typography)>
</WithStyles(Toolbar)>,
      "position": "static",
      "style": Object {
        "background": "rgba(10,191,82,1) linear-gradient(45deg, rgba(10,191,82,1) 0%, rgba(106,208,155,1) 100%)",
        "left": 0,
        "position": "fixed",
        "top": 0,
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(IconButton)
            aria-label="Menu"
            color="inherit"
>
            <pure(Menu) />
</WithStyles(IconButton)>,
          <WithStyles(Typography)
            color="inherit"
            variant="title"
>
            GoOut Adyen
</WithStyles(Typography)>,
        ],
        "variant": "dense",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "aria-label": "Menu",
            "children": <pure(Menu) />,
            "color": "inherit",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "GoOut Adyen",
            "color": "inherit",
            "variant": "title",
          },
          "ref": null,
          "rendered": "GoOut Adyen",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Toolbar)
          variant="dense"
>
          <WithStyles(IconButton)
                    aria-label="Menu"
                    color="inherit"
          >
                    <pure(Menu) />
          </WithStyles(IconButton)>
          <WithStyles(Typography)
                    color="inherit"
                    variant="title"
          >
                    GoOut Adyen
          </WithStyles(Typography)>
</WithStyles(Toolbar)>,
        "position": "static",
        "style": Object {
          "background": "rgba(10,191,82,1) linear-gradient(45deg, rgba(10,191,82,1) 0%, rgba(106,208,155,1) 100%)",
          "left": 0,
          "position": "fixed",
          "top": 0,
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(IconButton)
              aria-label="Menu"
              color="inherit"
>
              <pure(Menu) />
</WithStyles(IconButton)>,
            <WithStyles(Typography)
              color="inherit"
              variant="title"
>
              GoOut Adyen
</WithStyles(Typography)>,
          ],
          "variant": "dense",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "aria-label": "Menu",
              "children": <pure(Menu) />,
              "color": "inherit",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "GoOut Adyen",
              "color": "inherit",
              "variant": "title",
            },
            "ref": null,
            "rendered": "GoOut Adyen",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
